### ISSUE TITLE
Codec[String] is not being able to decode a String value.

### ISSUE THREAD

**Issue by jupco** [typelevel/vulcan#64]
I have an issue when trying to decode an Avro message published in Kafka that was encoded with Vulcan's help, when consuming the message the field type is `String` as, I presume, it was inferred by Magnolia for the given `case class`, but then the code of the `Codec[String]` has a case in the pattern matching that rejects everything that it's not an `Utf8` 

https://github.com/ovotech/vulcan/blob/master/modules/core/src/main/scala/vulcan/Codec.scala#L1796

I have provided a similar implicit codec that changes the type in the matching from `Utf8` to `String` and it worked, so is this a bug and should I propose a PR? or am I doing something wrong?

I did try putting this config ("avro.java.string" as "String") in the `AvroSettings` and didn't work.

Thanks for all the work you have put into Vulcan!

**Comment by vlovgr** [comment]
Is there any way you can provide a minimal example/test which shows the issue? It's not entirely unreasonable to also accept `String` when decoding, but would be good to understand why we need to do so with an example.

**Comment by jupco** [comment]
sorry, it's not that easy to come with a simple example (too much dependencies), in the meantime, I had been digging around this and found that this issue only affects the `key` deserialization, not the value's one, which is weird because both use the same `Codec[String]` at the end, I tried to change the key to a `Utf8` but there is no instance for a `Codec[Utf8]`

This is the stacktrace for the exception
```
vulcan.AvroException: Got unexpected type java.lang.String while decoding String, expected type Utf8
	at vulcan.AvroException$.apply(AvroException.scala:27)
	at vulcan.AvroError$$anon$1.throwable(AvroError.scala:46)
	at fs2.kafka.vulcan.AvroDeserializer$.$anonfun$using$4(AvroDeserializer.scala:38)
```

I'll try to come back with a simple test.

**Comment by vlovgr** [comment]
It indeed looks like the Confluent deserializer has a special case for `String`, where it's always returned as `String` rather than `Utf8`, as seen [here](https://github.com/confluentinc/schema-registry/blob/c4b487923e0193db5083b17f2744838edd8b24e1/avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java#L289-L290). I've verified this for both keys and values in fs2-kafka-vulcan. Will raise a fix and get a new release out.

**Comment by jupco** [comment]
Thanks!

**Comment by vlovgr** [comment]
[v0.2.2](https://github.com/ovotech/vulcan/releases/tag/v0.2.2) is out with the fix in #65. Thanks for raising this issue!
